### PR TITLE
search-652 separate revisit messages and put it in other queue, fixed…

### PR DESCRIPTION
**Pull request by lkalinka**
… logging of queu names

### CODE REVIEW

File: Command/CrawlCommand.php [hunk 1]
```diff
@@ -176,7 +179,7 @@ public function crawlUrl(AMQPMessage $message)
             ]);
             $this->spider->crawl($crawlJob, $this->queueFactory, $this->currentQueueType);
 
-            $this->logMessage('info', sprintf("Crawling %s", $crawlJob->getUrl()), $crawlJob->getUrl(), $data['metadata']['core']);
+            $this->logMessage('info', sprintf("Crawling %s", $crawlJob->getUrl()), $crawlJob->getUrl(), $data['metadata']['core'], $crawlJob->getQueueName());
```
Comment (websid) [inline Command/CrawlCommand.php]: line became to long and messy, can you split it up?

File: CrawlJob.php [hunk 2]
```diff
@@ -32,21 +32,28 @@ class CrawlJob
...
      *
      * @var string $url
      * @var string $baseUrl
      * @var array  $blacklist
      * @var array  $metadata
      * @var array  $whitelist
+     * @var null|string  $queueName
      */
-    public function __construct($url, $baseUrl, array $blacklist = [], array $metadata = [], array $whitelist = [])
+    public function __construct($url, $baseUrl, array $blacklist = [], array $metadata = [], array $whitelist = [], $queueName = null)
```
Comment (websid) [inline CrawlJob.php]: line became to long and messy, can you split it up?

File: Queue.php [hunk 3]
```diff
@@ -56,7 +63,13 @@ protected function getChannel()
     {
         if (null === $this->channel) {
             $this->channel = $this->connection->channel();
-            $this->channel->queue_declare($this->queueName, false, false, false, false);
+
+            $this->declareQueue($this->queueName);
+
+            if ($this->revisitDisabled == false) {
+                $this->declareQueue('revisit');
```
Comment (websid) [inline Queue.php:70]: can you cover it by unit tests?

File: Queue.php [hunk 4]
```diff
@@ -98,15 +121,11 @@ public function listen(Callable $callback)
...
-            false,
-            false,
-            false,
-            false,
-            $callback
-        );
+        $this->consumeBasic($this->queueName, $callback, $channel);
+
+        if ($this->revisitDisabled == false) {
+            $this->consumeBasic('revisit', $callback, $channel);
```
Comment (websid) [inline Queue.php:127]: can you cover it by unit tests?

File: Tests/CrawlJobTest.php [hunk 5]
```diff
@@ -44,7 +61,42 @@ public function ifGetBaseUrlReturnCorrectValue()
...
+        $this->assertSame('http://dummy.nl/dummy.html', $this->crawlJob->getUrl());
+    }
+
+    /**
+     * @test
+     * @testdox Tests if get blacklist returns correct value.
+     */
+    public function ifGetBlacklistReturnCorrectValue()
+    {
+        $this->assertSame(['http:/black.site/'], $this->crawlJob->getBlacklist());
```
Comment (websid) [inline Tests/CrawlJobTest.php]: missing / ;p

File: Queue.php (hunk 3 above)
Comment (lkalinka) [reply Queue.php:70]: Its covered there https://github.com/simgroep/concurrent-spider-bundle/pull/98/files#diff-3a9e0d716c811519f409dd6ea6e321adR124 and there https://github.com/simgroep/concurrent-spider-bundle/pull/98/files#diff-3a9e0d716c811519f409dd6ea6e321adR86

**Pull request merged**